orttraining/orttraining/python/training/ortmodule/__init__.py

```diff
@@ -15,6 +15,7 @@
 from onnxruntime.capi._pybind_state import is_ortmodule_available
 
 from ._fallback import ORTModuleFallbackException, ORTModuleInitException, _FallbackPolicy, wrap_exception
+from .options import _MemoryOptimizationLevel
```

Comment:
This comment is not specific to this line of code.

From your PR description:
1. Using checkpoint function is a strong indicator that the model is too large to fit in GPU memory.
2. But still there is a chance user models use checkpoint function for non-layerwise gradient checkpointing, which we don't think is a common use case.
3. On the other hand, if we don't override the checkpoint function here, mostly ONNX export will be failed.

I agree with your point 1 and 3. 
For 2, yes, non-layerwise checkpoint is not common, but in case some user has this usage, maybe it is better to have a way for user to disable ORT layer-wise-recompute?

Reply:
Agree. I have add following checks: 

`    The actual check is in the overridden function, verifying if:
    1) `checkpoint` is called during ORTModule model export,
    2) Gradient checkpoint autograd function is disallowed (ORTMODULE_ALLOW_AUTOGRAD_CHECKPOINT).
    If true, we reset memory optimization to layer-wise recompute.`

If user explicitly defined `ORTMODULE_MEMORY_OPT_LEVEL`, we will respect it instead of auto enabling it. In this way, they can choose to turn it ON if they found case 2.